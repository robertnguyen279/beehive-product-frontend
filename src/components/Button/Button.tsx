import React from 'react';
import { ButtonType } from 'types';

const Button = ({ className, children, onClick, isLoading, type, ...config }: ButtonType): React.ReactElement => {
  return (
    <button
      className={`button${isLoading ? '--loading' : ''} text-sm text-white cursor-pointer ${
        className ? className : ''
      }`}
      onClick={onClick}
      type={type}
      {...config}
    >
      {children}
    </button>
  );
};

export default Button;
