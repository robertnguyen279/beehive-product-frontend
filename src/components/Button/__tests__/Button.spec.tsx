import React from 'react';
import { shallow } from 'enzyme';

import Button from '../Button';
const clickFn = jest.fn();

describe('Loading', () => {
  const component = shallow(<Button onClick={clickFn}>Simulate button</Button>);
  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });
  it('should been clicked', () => {
    component.find('button').simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
});
