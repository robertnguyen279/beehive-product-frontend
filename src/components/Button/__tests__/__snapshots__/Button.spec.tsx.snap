// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `
<button
  className="button text-sm text-white "
  onClick={[MockFunction]}
>
  Simulate button
</button>
`;
